### PR TITLE
adds optional $message parameter

### PR DESCRIPTION
**Pull request by ibronit**
It makes possible to provide custom exception message to the exception.

### CODE REVIEW

File: lib/Doctrine/ORM/NoResultException.php [hunk 1]
```diff
@@ -9,11 +9,13 @@
  */
 class NoResultException extends UnexpectedResultException
 {
+    public const DEFAULT_MESSAGE = 'No result was found for query although at least one row was expected.';
```
Comment (Ocramius) [inline lib/Doctrine/ORM/NoResultException.php]: This exception is not supposed to be thrown by userland code.

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
Closing as `invalid`: please define your own exception types.

**Comment by smatyas** [comment]
@Ocramius so why does this file have the same feature?
https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/NonUniqueResultException.php

**Comment by Ocramius** [comment]
@smatyas probably shouldn't, tbh. /cc @greg0ire

**Comment by Majkl578** [comment]
Also see #6743.

**Pull request closed without merging**